astropy/cosmology/io.py

```diff
@@ -0,0 +1,351 @@
+
+        # make cosmo instance
+        ba = cosmo_cls._init_signature.bind_partial(**params, meta=meta)
+        cosmo = cosmo_cls(*ba.args, **ba.kwargs)
+        return cosmo
+
+    def from_table(self, table, index=None):
+        """Instantiate a `~astropy.cosmology.Cosmology` from a |QTable|.
+
+        Parameters
+        ----------
+        table : `~astropy.QTable`
+        index : int or None, optional
+            The row from table.
+            TODO! allow for index to be a string -- the name fromo the row.
+
+        Returns
+        -------
+        `~astropy.cosmology.Cosmology`
+
+        """
+        # get 1-row of table
+        row = table.iloc[index] if index is not None else table
+        if len(row) != 1:  # check that it's one row
+            raise IncompatibleShapeError("Table must be 1-D.")
+
+        # special values
+        # get name from 1) column 2) metadata 3) it's None
+        name = row.columns.get("name", [row.meta.pop("name", None)])[0]
+        # metadata is a mix of the general metadata and row-specific data
+        meta = {k: v for k, v in row.meta.items() if not isinstance(k, int)}
+        meta.update(row.meta.get(index, {}))
+        # the cosmology class is in the table's metadata
+        cosmo_cls = meta.pop("cosmology", None)
+
+        # turn row into mapping (dict of the arguments)
+        mapping = {k: v[0] for k, v in zip(row.colnames, row.values())}
+        mapping["name"] = name
+        mapping["meta"] = {**meta, **mapping.get("meta", {})}
+        mapping.setdefault("cosmology", cosmo_cls)
+
+        # build cosmology from map
+        cosmo = self.from_mapping(mapping)
+        return cosmo
+
+
+class CosmologyWrite(registry.UnifiedReadWrite):
+    """Write this Cosmology object out in the specified format.
+
+    This function provides the Table interface to the astropy unified I/O
+    layer.  This allows easily writing a file in many supported data formats
+    using syntax such as::
+
+      >>> from astropy.table import Table
+      >>> dat = Table([[1, 2], [3, 4]], names=('a', 'b'))
+      >>> dat.write('table.dat', format='ascii')
+
+    Get help on the available writers for ``Table`` using the``help()`` method::
+
+      >>> Table.write.help()  # Get help writing Table and list supported formats
+      >>> Table.write.help('fits')  # Get detailed help on Table FITS writer
+      >>> Table.write.list_formats()  # Print list of available formats
+
+    The ``serialize_method`` argument is explained in the section on
+    `Table serialization methods
+    <https://docs.astropy.org/en/latest/io/unified.html#table-serialization-methods>`_.
+    See also: https://docs.astropy.org/en/stable/io/unified.html
+
+    Parameters
+    ----------
+    *args : tuple, optional
+        Positional arguments passed through to data writer. If supplied the
+        first argument is the output filename.
+    format : str
+        File format specifier.
+    serialize_method : str, dict, optional
```

Comment:
Why is `serialize_method` specifically exposed here? Seems like that is not really needed.

Reply:
Most of the docs were stolen from the Table I/O docs.

Reply:
Thanks. I've removed it as I prune the docstring and add more relevant documentation.